### PR TITLE
Add a new integration test for "Custom Instrumentation Telemetry RFC"

### PR DESCRIPTION
**Pull request by mcculls**
## Motivation

Add a new integration test for "Custom Instrumentation Telemetry RFC"

## Changes

Tests the following changes from the "Custom Instrumentation Telemetry RFC" on Java:
* OpenTelemetry drop-in libraries have integration names starting with 'otel.'
* Spans created by drop-in libraries are tagged with names starting with 'otel.'

Also includes some fixes due to R2DBC syntax differing slightly from JDBC

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: tests/integrations/test_otel_drop_in.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Unless explicitly stated otherwise all files in this repository are licensed under the the Apache License Version 2.0.
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2024 Datadog, Inc.
+
+from utils import weblog, interfaces, context, missing_feature, scenarios, features
+
+
+@features.f_otel_interoperability
+@missing_feature(condition=context.library != "java", reason="Endpoint is not implemented on weblog")
```
Comment (robertomonteromiguel) [inline tests/integrations/test_otel_drop_in.py:9]: Could we define this in the manifest files?

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]

File: tests/integrations/test_otel_drop_in.py (hunk 1 above)
Comment (mcculls) [reply tests/integrations/test_otel_drop_in.py]: Yes, good point: a7415c75f

File: utils/build/docker/java/install_drop_in.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+set -eu
+
+if [ $(ls /binaries/opentelemetry-javaagent-r2dbc*.jar | wc -l) = 0 ]; then
+    LIBRARY_URL='https://repo1.maven.org/maven2/io/opentelemetry/javaagent/instrumentation/opentelemetry-javaagent-r2dbc-1.0/2.5.0-alpha/opentelemetry-javaagent-r2dbc-1.0-2.5.0-alpha.jar'
+    echo "install from Maven central: $LIBRARY_URL"
```
Comment (cbeauchesne) [inline utils/build/docker/java/install_drop_in.sh:7]: Could you document in `docs/execute/binaries.md` what are the possible options ?

**Review by cbeauchesne — COMMENTED** [review]
Small doc request, then all good from framework usage.

Could you request a review from someone familiar with the feature ?

File: utils/build/docker/java/install_drop_in.sh (hunk 2 above)
Comment (mcculls) [reply utils/build/docker/java/install_drop_in.sh:7]: Documented in 1cea91c12

File: docs/execute/binaries.md [hunk 3]
```diff
@@ -83,6 +83,17 @@ To run a custom tracer version from a remote branch:
 2. Open the details of the test in CircleCi and click on the `Artifacts` tab.
 3. Download the `libs/dd-java-agent-*-SNAPSHOT.jar` and `libs/dd-trace-api-*-SNAPSHOT.jar` and move them into the `system-tests/binaries/` folder.
 4. Follow Step 4 from above to run the Parametric tests.
+
+Follow these steps to run the OpenTelemetry drop-in test with a custom drop-in version:
```
Comment (cbeauchesne) [inline docs/execute/binaries.md:87]:
```suggestion
### Open Telemetry

Follow these steps to run the OpenTelemetry drop-in test with a custom drop-in version:
```

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Small doc improvment

**Review by emmettbutler — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]
LGTM with one open question about the telemetry metrics test case

File: tests/integrations/test_otel_drop_in.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+        integration_found = False
+        for data in interfaces.library.get_telemetry_data():
+            payload = data["request"]["content"].get("payload")
+            if payload and has_otel_integration(payload.get("integrations", [])):
+                integration_found = True
+                break
+        assert integration_found, "Otel drop-in telemetry data not found"
+
+    def test_otel_drop_in_span_metrics(self):
```
[the remark this answers is not in the document]
Comment (mcculls) [reply tests/integrations/test_otel_drop_in.py:32]: The single setup method is enough to populate the data used by both test methods, but I updated it to be more consistent with other test classes and share a common setup across all test methods: eb9bc2b

**Pull request merged**